Source/Phone/Call/CallManager.swift

```diff
@@ -89,35 +100,42 @@ class CallManager {
         
         // If it belongs to existing active call, update it.
         if let call = callInstances[callUrl] {
-			call.update(callInfo: callInfo)
-            return
-        }
-        
-        if callInfo.isIncomingCall {
-            doActionWhenIncoming(callInfo)
-        } else if callInfo.hasJoinedOnOtherDevice {
-            doActionWhenJoinedOnOtherDevice(callInfo)
+            call.update(callInfo: callInfo)
+        } else if let deviceUrlString = deviceService.deviceUrl,
+            let deviceUrl = URL(string: deviceUrlString),
+            callInfo.isIncomingCall,
+            callInfo.hasJoinedOnOtherDevice(deviceUrl: deviceUrl) {
```

Comment:
That looks screwy. We'll take a hard look at that and see what might be the problem, but there's definitely semantically a problem there.